src/main/java/com/jnape/palatable/lambda/monad/transformer/builtin/IterateT.java

```diff
@@ -289,34 +279,16 @@ private IterateT(Pure<M> pureM,
         return MonadT.super.discardR(appB).coerce();
     }
 
-    private MonadRec<RecursiveResult<IterateT<M, A>, Maybe<Tuple2<A, IterateT<M, A>>>>, M> resume() {
-        return conses.head().match(
-                __ -> middles.head().match(
-                        ___ -> snocs.head().match(
-                                ____ -> pureM.apply(terminate(nothing())),
-                                ma -> ma.fmap(a -> terminate(just(tuple(a, new IterateT<>(pureM,
-                                                                                          snocs.tail(),
-                                                                                          ImmutableQueue.empty(),
-                                                                                          ImmutableQueue.empty())))))),
-                        lazyOrStrict -> lazyOrStrict.match(
-                                lazy -> lazy.apply().fmap(maybeRes -> maybeRes.match(
-                                        ___ -> recurse(new IterateT<>(pureM, ImmutableQueue.empty(), middles.tail(), snocs)),
-                                        into((a, as) -> recurse(new IterateT<>(pureM,
-                                                                               ImmutableQueue.singleton(pureM.apply(a)),
-                                                                               ImmutableQueue.singleton(b(migrateForward(as))),
-                                                                               ImmutableQueue.empty())))
-                                )),
-                                strict -> pureM.apply(recurse(migrateForward(strict))))),
-                ma -> ma.fmap(a -> terminate(just(tuple(a, new IterateT<>(pureM, conses.tail(), middles, snocs))))));
-    }
-
-    private IterateT<M, A> migrateForward(IterateT<M, A> as) {
-        if (middles.tail().isEmpty()) {
-            return new IterateT<>(pureM, conses.concat(as.conses), as.middles, as.snocs.concat(snocs));
-        }
-
-        IterateT<M, A> lasts = new IterateT<>(pureM, as.snocs, middles.tail(), snocs);
-        return new IterateT<>(pureM, as.conses, as.middles.pushBack(b(lasts)), ImmutableQueue.empty());
+    private static <M extends MonadRec<?, M>, A>
+    Fn1<ImmutableQueue<Choice2<Fn0<MonadRec<Maybe<Tuple2<A, IterateT<M, A>>>, M>>, MonadRec<A, M>>>, MonadRec<RecursiveResult<ImmutableQueue<Choice2<Fn0<MonadRec<Maybe<Tuple2<A, IterateT<M, A>>>, M>>, MonadRec<A, M>>>, Maybe<Tuple2<A, IterateT<M, A>>>>, M>>
```

Comment:
My kingdom for an alias. Any preferences for where to break this line?

Reply:
Meh, break between input and output of the function I guess.